packages/prop-types/src/array-of-max-length.js

```diff
@@ -0,0 +1,45 @@
+import PropTypes from 'prop-types'
+
+import PropTypeError from './prop-type-error'
+import { createChainableTypeChecker, getCircularReplacer } from './utils'
+
+export default function arrayOfMaxLength(
```

Comment:
This is the right solution.  But given the other issue that we've observed, where proptype code isn't being removed from builds (eg, elementOfType), this is a lot of code for validating proptypes, when we could just use `Array.slice` to limit what's displayed.  So, I'm not sure what's best at the moment, considering the bundle size tension here.  Thoughts?

Reply:
oh i wasn't aware that the code isn't be eliminated. i thought about using slice but the consumer of the component would just see something different than expected(ie only two items)...i feel like a warning is more useful/actionable